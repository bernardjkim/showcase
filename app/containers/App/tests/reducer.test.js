import { fromJS } from 'immutable';
import appReducer from '../reducer';

import {
  createToken,
  createTokenError,
  createTokenSuccess,
  deleteToken,
  loadUser,
  loadUserError,
  loadUserSuccess,
} from '../actions';

describe('appReducer', () => {
  let state;
  beforeEach(() => {
    state = fromJS({
      loading: false,
      error: false,
      user: false,
      validateToken: true,
    });
  });
  it('returns the initial state', () => {
    expect(appReducer(undefined, {})).toMatchSnapshot();
  });

  it('handle the create token action', () => {
    const email = 'email';
    const password = 'password';
    expect(appReducer(state, createToken(email, password))).toMatchSnapshot();
  });

  it('handle the create token success action', () => {
    expect(appReducer(state, createTokenSuccess())).toMatchSnapshot();
  });

  it('handle the create token error action', () => {
    const error = new Error('Test error');
    expect(appReducer(state, createTokenError(error))).toMatchSnapshot();
  });

  it('handle the delete token action', () => {
    expect(appReducer(state, deleteToken())).toMatchSnapshot();
  });

  it('handle the load user action', () => {
    expect(appReducer(state, loadUser())).toMatchSnapshot();
  });

  it('handle the load user success action', () => {
    const user = {};
    expect(appReducer(state, loadUserSuccess(user))).toMatchSnapshot();
  });

  it('handle the load user error action', () => {
    const error = new Error('Test error');
    expect(appReducer(state, loadUserError(error))).toMatchSnapshot();
  });
});
