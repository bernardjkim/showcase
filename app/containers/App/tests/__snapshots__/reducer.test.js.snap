// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appReducer handle the create token action 1`] = `
Immutable.Map {
  "loading": true,
  "error": false,
  "user": false,
  "validateToken": true,
}
`;

exports[`appReducer handle the create token error action 1`] = `
Immutable.Map {
  "loading": false,
  "error": [Error: Test error],
  "user": false,
  "validateToken": true,
}
`;

exports[`appReducer handle the create token success action 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "user": false,
  "validateToken": true,
}
`;

exports[`appReducer handle the delete token action 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "user": false,
  "validateToken": true,
}
`;

exports[`appReducer handle the load user action 1`] = `
Immutable.Map {
  "loading": true,
  "error": false,
  "user": false,
  "validateToken": false,
}
`;

exports[`appReducer handle the load user error action 1`] = `
Immutable.Map {
  "loading": false,
  "error": [Error: Test error],
  "user": false,
  "validateToken": true,
}
`;

exports[`appReducer handle the load user success action 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "user": Immutable.Map {},
  "validateToken": true,
}
`;

exports[`appReducer returns the initial state 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
  "user": false,
  "validateToken": true,
}
`;
